### PR TITLE
PantheonTerminalWindow.vala: make constructor properties public

### PR DESCRIPTION
**Pull request by danrabbit**
* make `app`, `restore_pos`, and `recreate_tabs` public properties
* Sort public and private variables
* Use `Object` instead of passing vars to `init`
* Combine `init` and `construct`

### CODE REVIEW

File: src/PantheonTerminalWindow.vala [hunk 1]
```diff
@@ -53,9 +39,19 @@ namespace PantheonTerminal {
         private const string SOLARIZED_LIGHT_BG = "rgba(253, 246, 227, 0.95)";
         private const string SOLARIZED_LIGHT_FG = "#586e75";
 
-
-
+        public bool unsafe_ignored;
+        public bool focus_restored_tabs { get; construct; default = true; }
+        public bool recreate_tabs { get; construct; default = true; }
+        public bool restore_pos { get; construct; default = true; }
+        public PantheonTerminalApp app { get; construct; }
```
Comment (jeremypw) [inline src/PantheonTerminalWindow.vala]: This duplicates Widget.application - is it advantageous to have a separate property as well as set_application (app) in construct {}?

File: src/PantheonTerminalWindow.vala [hunk 2]
```diff
@@ -53,9 +39,19 @@ namespace PantheonTerminal {
...
         public SimpleActionGroup actions { get; construct; }
+        
+        public GLib.List <TerminalWidget> terminals = new GLib.List <TerminalWidget> ();
+        public Granite.Widgets.DynamicNotebook notebook;
```
Comment (jeremypw) [inline src/PantheonTerminalWindow.vala]: Can be made private.

File: src/PantheonTerminalWindow.vala [hunk 3]
```diff
@@ -53,9 +39,19 @@ namespace PantheonTerminal {
...
+        public Gtk.ActionGroup main_actions;
```
Comment (jeremypw) [inline src/PantheonTerminalWindow.vala]: Can be made read-only with { get; private set; }

File: src/PantheonTerminalWindow.vala [hunk 4]
```diff
@@ -53,9 +39,19 @@ namespace PantheonTerminal {
...
+        public Gtk.ToggleButton search_button;
+        public Gtk.UIManager ui;
```
Comment (jeremypw) [inline src/PantheonTerminalWindow.vala]: Can be made read-only with { get; private set; }

File: src/PantheonTerminalWindow.vala [hunk 5]
```diff
@@ -53,9 +39,19 @@ namespace PantheonTerminal {
...
+        public TerminalWidget current_terminal = null;
```
Comment (jeremypw) [inline src/PantheonTerminalWindow.vala]: Can be made read-only with { get; private set; }

File: src/PantheonTerminalWindow.vala [hunk 6]
```diff
@@ -53,9 +39,19 @@ namespace PantheonTerminal {
...
+        public bool focus_restored_tabs { get; construct; default = true; }
+        public bool recreate_tabs { get; construct; default = true; }
+        public bool restore_pos { get; construct; default = true; }
+        public PantheonTerminalApp app { get; construct; }
         public SimpleActionGroup actions { get; construct; }
+        
+        public GLib.List <TerminalWidget> terminals = new GLib.List <TerminalWidget> ();
+        public Granite.Widgets.DynamicNotebook notebook;
+        public Gtk.ActionGroup main_actions;
+        public Gtk.ToggleButton search_button;
```
Comment (jeremypw) [inline src/PantheonTerminalWindow.vala]: Can be made private

File: src/PantheonTerminalWindow.vala (hunk 2 above)
Comment (jeremypw) [inline src/PantheonTerminalWindow.vala]: Can be made private

Comment (jeremypw) [inline src/PantheonTerminalWindow.vala]: Can be made private

File: src/PantheonTerminalWindow.vala (hunk 1 above)
Comment (danrabbit) [reply src/PantheonTerminalWindow.vala]: I'd like to chase this down in a follow-up PR, but there seems to be some conflict between `Gtk.Application` and `PantheonTerminalApp`. Trying to use `Widget.Application` is giving me errors about setting shortcut accelerators and some other issues.

**Pull request merged**